lib/internal/process/per_thread.js

```diff
@@ -295,15 +298,15 @@ function buildAllowedFlags() {
   const nodeFlags = new SafeSet(ArrayPrototypeMap(allowedNodeEnvironmentFlags,
                                                   trimLeadingDashes));
 
-  class NodeEnvironmentFlagsSet extends SafeSet {
-    constructor(...args) {
-      super(...args);
+  class NodeEnvironmentFlagsSet extends Set {
+    #set;
+    constructor(iterable) {
+      super();
+      this.#set = new SafeSet(iterable);
+    }
 
-      // The super constructor consumes `add`, but
-      // disallow any future adds.
-      ObjectDefineProperty(this, 'add', {
-        value: () => this
-      });
+    add() {
+      // No-op
```

Comment:
Nit: perhaps using this comment is more helpful here
https://github.com/nodejs/node/blob/fd02dace6ffa33b21d6bbb9f1d026c8a931c7594/lib/internal/process/per_thread.js#L311

Reply:
```suggestion
      // No-op, `Set` API compatible
```